[Fleet] Remove `upgradePackage` and consolidate it with `installPackage`, optimize calls to create index patterns

jen-huang opened the pull request:
## Summary

~~This PR resolves Fleet plugin's portion of #91715 by removing direct calls to index pattern saved objects in `server/services/epm/kibana/index_pattern/install.ts` and replacing them with calls to data plugin's `IndexPatternService`.~~

~~Changes were made to data plugin's `IndexPatternService` to allow non-scripted fields to be returned from `getAsSavedObjectBody` when `allowNoIndex` is true: afc91ce. This allows the call to [create and save](https://github.com/elastic/kibana/blob/69b93da1807cc9a613be1ad75b37df6bcee8d676/src/plugins/data/common/index_patterns/index_patterns/index_patterns.ts#L508) the index pattern to work correctly.~~

**March 17 update:**
* This PR no longer has any change to the way EPM installs index patterns, the current `IndexPatternService` does not work for the way EPM installs index patterns. We decided to pull out the changes I had here to get the service working for EPM in favor of having more discussions about potential solutions first, so unfortunately we still need to write directly to the index pattern saved objects themselves in the meantime 😞 
* However, this PR still sets us up to easily swap out SO for `IndexPatternService` when that's ready. Plus I think some of the simplifications introduced in this PR are beneficial for our error handling efforts, so I'd like to still get these changes in if the team agrees.

This PR introduces the following changes to EPM package installation:
- Removes `upgradePackage` method, which was only being used by `bulkInstallPackages`, since `installPackage` can do the same thing (upgrade a package)
  - Adjusts the code in `bulkInstallPackages` to use `installPackage` instead
  - Preserve `upgradePackage`'s use of `handleInstallPackageFailure` by adding it to `installPackageFromRegistry`
- Adds new type `InstallResult` and use in various places where we previously had just `AssetReference[]`, this new type will allow us to be more verbose about install results in the future, for now it includes an additional `status: 'installed' | 'already_installed'` property
- Runs `installIndexPatterns` as a skippable post-install operation in `installPackage`, rather than in `_installPackage()`
  - This allows `bulkInstallPackages` to skip this operation and run `installIndexPatterns` after all package installations are done to avoid race conditions that create duplicate index patterns (for example, on plugin setup call where we install multiple default packages, multiple `logs-*` and `metrics-*` patterns were created)
  - ~~The race condition was not an issue before when we were overwriting directly to SO based on ID~~ We are keeping writing to SOs, but this still reduces the the amount of times we write to them
- Removes our custom index pattern field typings in favor of the ones from data plugin and adjust our field generation code to return the correct field format
  - `analyzed` and `doc_value` properties were removed for _index pattern_ fields after offline discussions, they are not used by Kibana, they are preserved for _index template_ fields
  - `type` will default to `string` for unknown types since `IndexPatternService` does not support undefined `type`
- Adds logging around modified areas

Even though there are quite a lot of changes here, everything around package installation and index patterns should behave the same way as before. To test this PR, ensure that the index patterns are created properly, with the right fields, whenever packages are installed, ungraded, or removed.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112870/)
* Commit: 0890eaa50f5ed5af1c20d38ec5991c81e7359c17
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112870/execution/node/432/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @jen-huang

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112870/","number":"112870","commit":"0890eaa50f5ed5af1c20d38ec5991c81e7359c17"}]}
PIPELINE-->

elasticmachine commented:
Pinging @elastic/fleet (Team:Fleet)

jen-huang commented:
@mattkime I know we are pending discussion for the problem described in #94512, but what do you think about this subset of changes from that PR to enable Fleet's use of `createAndSave()`? I think this change can be considered a separate problem/solution:

> In addition, changes were made to data plugin's `IndexPatternService` to allow non-scripted fields to be returned from `getAsSavedObjectBody` when `allowNoIndex` is true: afc91ce. This allows the call to [create and save](https://github.com/elastic/kibana/blob/69b93da1807cc9a613be1ad75b37df6bcee8d676/src/plugins/data/common/index_patterns/index_patterns/index_patterns.ts#L508) the index pattern to work correctly.

jen-huang commented:
@skh FYI on the changes to package installation here. I know you are working on a related area re: handling upgrade failures, I tried to focus on simplification in this PR, so I hope these changes are actually beneficial for that work. Let me know if there is anything in this PR that conflicts with your work or is a concern.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113502/)
* Commit: c2bf522973600a1192a8c485abe68922e5600fe0
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-94490/c2bf522973600a1192a8c485abe68922e5600fe0)
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroup7](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113502/execution/node/778/log)
* [Execute xpack-kibana-ciGroup7](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113502/execution/node/935/log)

### History
* :broken_heart: [Build #112921](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112921/) failed 8e712e9c0087fc7f9e2b3dc0c729a23f0265f243
* :broken_heart: [Build #112870](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112870/) failed 0890eaa50f5ed5af1c20d38ec5991c81e7359c17

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @jen-huang

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112870/","number":"112870","commit":"0890eaa50f5ed5af1c20d38ec5991c81e7359c17"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112921/","number":"112921","commit":"8e712e9c0087fc7f9e2b3dc0c729a23f0265f243"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113502/","number":"113502","commit":"c2bf522973600a1192a8c485abe68922e5600fe0"}]}
PIPELINE-->

jfsiii commented:
cc @jonathan-buttner for 👀. If I recall correctly, some of the updatePackage and bulkInstallPackage code was added/updated for certain scenarios with Endpoint. If so, can you help test them and make sure they still work?

jonathan-buttner commented on x-pack/plugins/fleet/server/services/epm/packages/bulk_install_packages.ts:
```diff
@@ -7,58 +7,71 @@
+  // only install index patterns if we installed any package for the first time
+  if (
+    installResults.find(
+      (result) => result.status === 'fulfilled' && result.value.result?.status === 'installed'
```
Couple questions:

1. I haven't looked thoroughly at how installIndexPatterns works but if an upgrade of a package includes new mapping changes will those get applied if we don't call `installIndexPatterns` here? Would we not want to update the index patterns with the new mappings from the package?
2. Will the `status` field have values other than `installed`?

When I look at this code: https://github.com/elastic/kibana/blob/c91092fffdf954b05b7089efe219e9e7de24fbe0/x-pack/plugins/fleet/server/services/epm/packages/install.ts#L249

wouldn't the return value in a successful scenario of an upgrade or an install always be `installed`?

So I think this would return true if a package was installed for the first time or upgraded right?

jen-huang replied:
`status` can be `installed` or `already_installed`. When a package is successfully upgraded, the status will be `installed`, so this condition will be true, meaning that we will install index patterns. It only returns `already_installed` when the same package version is requested to be installed: https://github.com/elastic/kibana/blob/c91092fffdf954b05b7089efe219e9e7de24fbe0/x-pack/plugins/fleet/server/services/epm/packages/install.ts#L209

jonathan-buttner replied:
Oh I see, sorry for the misunderstanding!

jonathan-buttner commented on x-pack/test/fleet_api_integration/apis/epm/bulk_upgrade.ts:
```diff
@@ -51,8 +51,7 @@ export default function (providerContext: FtrProviderContext) {
         const entry = body.response[0] as BulkInstallPackageInfo;
-        expect(entry.oldVersion).equal('0.1.0');
-        expect(entry.newVersion).equal('0.3.0');
+        expect(entry.version).equal('0.3.0');
```
I'm fine with removing `oldVersion` as the SecuritySolution app doesn't use it but the intent behind it was to be able to identify when an install for the first time occurs vs an upgrade of a package.

jen-huang replied:
Interesting, I removed this since it was logic that was specific to the now-removed `upgradePackage` method and to add it to `installPackage` would mean additional overhead. Maybe we can leverage the new `status` field to provide more granularity about what type of install happened (currently it's a blanket `install`), even if we don't return the actual "old version" info.

jonathan-buttner commented:
> cc @jonathan-buttner for 👀. If I recall correctly, some of the updatePackage and bulkInstallPackage code was added/updated for certain scenarios with Endpoint. If so, can you help test them and make sure they still work?

Yeah when a user navigates to the Security Solution app we make a request into fleet to the upgrade api which will either install the endpoint package for the first time, or attempt to upgrade it to the latest version. I'll take a look tomorrow.

jen-huang commented:
Thanks @jonathan-buttner for looking, I see a couple of Endpoint tests failed so I might have broken something indeed, I'll debug tomorrow.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113522/)
* Commit: c91092fffdf954b05b7089efe219e9e7de24fbe0
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-94490/c91092fffdf954b05b7089efe219e9e7de24fbe0)
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroup7](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113522/execution/node/766/log)
* [Execute xpack-kibana-ciGroup7](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113522/execution/node/935/log)

### History
* :broken_heart: [Build #113502](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113502/) failed c2bf522973600a1192a8c485abe68922e5600fe0
* :broken_heart: [Build #112921](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112921/) failed 8e712e9c0087fc7f9e2b3dc0c729a23f0265f243
* :broken_heart: [Build #112870](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112870/) failed 0890eaa50f5ed5af1c20d38ec5991c81e7359c17

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @jen-huang

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112870/","number":"112870","commit":"0890eaa50f5ed5af1c20d38ec5991c81e7359c17"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112921/","number":"112921","commit":"8e712e9c0087fc7f9e2b3dc0c729a23f0265f243"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113502/","number":"113502","commit":"c2bf522973600a1192a8c485abe68922e5600fe0"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113522/","number":"113522","commit":"c91092fffdf954b05b7089efe219e9e7de24fbe0"}]}
PIPELINE-->

mattkime commented:
@jen-huang I'm skeptical but we can discuss it tomorrow.

skh commented:
> @skh FYI on the changes to package installation here. I know you are working on a related area re: handling upgrade failures, I tried to focus on simplification in this PR, so I hope these changes are actually beneficial for that work. Let me know if there is anything in this PR that conflicts with your work or is a concern.

Changes look good to me, good job on the simplification! I'll have to redo some bits from what I'm working on, but if this PR gets in reasonably quickly that's not a problem at all.

jonathan-buttner commented:
@jen-huang I tested installing the endpoint package using the `_bulk` API and upgrading it and both worked:

<details><summary>Installing the package for the first time</summary>

![image](https://user-images.githubusercontent.com/56361221/111528351-6b6a1f00-8737-11eb-9a65-4b2ac2041868.png)


</details>

<details><summary>Upgrading to 0.19.0</summary>

![image](https://user-images.githubusercontent.com/56361221/111528673-d3206a00-8737-11eb-852a-21f160d251aa.png)


</details>

jonathan-buttner commented on x-pack/plugins/fleet/server/services/epm/packages/install.ts:
```diff
@@ -247,32 +190,75 @@ async function installPackageFromRegistry({
-  if (semverLt(pkgVersion, latestPackage.version) && !force && !installOutOfDateVersionOk) {
-    throw new PackageOutdatedError(`${pkgkey} is out-of-date and cannot be installed or updated`);
+
+  const forceInstall = force || ['reinstall', 'reupdate', 'rollback'].includes(installType);
```
I think this is going to cause a reinstall of a package when the `_bulk` API is hit.

If the currently installed version is `1.0.0` I think `installType` will return `reinstall` which would result in `forceInstall` being `true`. The `_bulk` API is only passed a package name and it then requests the latest version. If the user already has the latest version installed would this reinstall it?

I think I noticed this in my testing because I saw logs for the transform being started and stopped each time the `_bulk` api is hit.

The `upgradePackage` function avoiding reinstalling a package when the current version equaled the latest version.

This could be problematic because a request to the `_bulk` API is made pretty frequently from the security solution app.

jen-huang replied:
thanks for spotting this and testing! I noticed this too when I was testing, it's a bug I introduced late yesterday 😅  I will fix in my next update

jen-huang replied:
pushed update that should fix this

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113863/)
* Commit: e534f709856e97e3d35dec508140f3d2abc6e6fe
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Setup Build Environment and Dependencies](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113863/execution/node/122/log)
* [Check Types](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113863/execution/node/458/log)
* [Setup Build Environment and Dependencies](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113863/execution/node/163/log)

### History
* :broken_heart: [Build #113522](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113522/) failed c91092fffdf954b05b7089efe219e9e7de24fbe0
* :broken_heart: [Build #113502](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113502/) failed c2bf522973600a1192a8c485abe68922e5600fe0
* :broken_heart: [Build #112921](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112921/) failed 8e712e9c0087fc7f9e2b3dc0c729a23f0265f243
* :broken_heart: [Build #112870](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112870/) failed 0890eaa50f5ed5af1c20d38ec5991c81e7359c17

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @jen-huang

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112870/","number":"112870","commit":"0890eaa50f5ed5af1c20d38ec5991c81e7359c17"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112921/","number":"112921","commit":"8e712e9c0087fc7f9e2b3dc0c729a23f0265f243"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113502/","number":"113502","commit":"c2bf522973600a1192a8c485abe68922e5600fe0"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113522/","number":"113522","commit":"c91092fffdf954b05b7089efe219e9e7de24fbe0"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113863/","number":"113863","commit":"e534f709856e97e3d35dec508140f3d2abc6e6fe"}]}
PIPELINE-->

jen-huang commented:
@elastic/fleet @jonathan-buttner Per discussion with @mattkime and others, we decided to table the changes to `IndexPatternService` for now, so this means EPM still needs to rely on writing directly to SOs. I've updated the code accordingly and updated [PR description](https://github.com/elastic/kibana/pull/94490#issue-591168295) with more details. Let me know if any Qs or concerns about the rest of the changes. Thank you!

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113869/)
* Commit: cc069bffdd5ad935df1eaf68697ddc5d0bec5d54
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-94490/cc069bffdd5ad935df1eaf68697ddc5d0bec5d54)
* This comment will update when the build is complete

### Failed CI Steps
* [Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113869/execution/node/171/log)

### History
* :broken_heart: [Build #113863](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113863/) failed e534f709856e97e3d35dec508140f3d2abc6e6fe
* :broken_heart: [Build #113522](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113522/) failed c91092fffdf954b05b7089efe219e9e7de24fbe0
* :broken_heart: [Build #113502](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113502/) failed c2bf522973600a1192a8c485abe68922e5600fe0
* :broken_heart: [Build #112921](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112921/) failed 8e712e9c0087fc7f9e2b3dc0c729a23f0265f243
* :broken_heart: [Build #112870](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112870/) failed 0890eaa50f5ed5af1c20d38ec5991c81e7359c17

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @jen-huang

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112870/","number":"112870","commit":"0890eaa50f5ed5af1c20d38ec5991c81e7359c17"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112921/","number":"112921","commit":"8e712e9c0087fc7f9e2b3dc0c729a23f0265f243"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113502/","number":"113502","commit":"c2bf522973600a1192a8c485abe68922e5600fe0"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113522/","number":"113522","commit":"c91092fffdf954b05b7089efe219e9e7de24fbe0"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113863/","number":"113863","commit":"e534f709856e97e3d35dec508140f3d2abc6e6fe"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113869/","number":"113869","commit":"cc069bffdd5ad935df1eaf68697ddc5d0bec5d54"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114050/)
* Commit: 54ff3ac46f9dc7ee2df690637aa5ed73cbc07bb8
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-94490/54ff3ac46f9dc7ee2df690637aa5ed73cbc07bb8)
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-firefox](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114050/execution/node/682/log)
* [Execute xpack-firefox](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114050/execution/node/814/log)

### History
* :broken_heart: [Build #113869](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113869/) failed cc069bffdd5ad935df1eaf68697ddc5d0bec5d54
* :broken_heart: [Build #113863](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113863/) failed e534f709856e97e3d35dec508140f3d2abc6e6fe
* :broken_heart: [Build #113522](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113522/) failed c91092fffdf954b05b7089efe219e9e7de24fbe0
* :broken_heart: [Build #113502](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113502/) failed c2bf522973600a1192a8c485abe68922e5600fe0
* :broken_heart: [Build #112921](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112921/) failed 8e712e9c0087fc7f9e2b3dc0c729a23f0265f243

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @jen-huang

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/112921/","number":"112921","commit":"8e712e9c0087fc7f9e2b3dc0c729a23f0265f243"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113502/","number":"113502","commit":"c2bf522973600a1192a8c485abe68922e5600fe0"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113522/","number":"113522","commit":"c91092fffdf954b05b7089efe219e9e7de24fbe0"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113863/","number":"113863","commit":"e534f709856e97e3d35dec508140f3d2abc6e6fe"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113869/","number":"113869","commit":"cc069bffdd5ad935df1eaf68697ddc5d0bec5d54"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114050/","number":"114050","commit":"54ff3ac46f9dc7ee2df690637aa5ed73cbc07bb8"}]}
PIPELINE-->

jen-huang commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114322/)
* Commit: f12ddfcb6a1b523c4523c2fac8ac5e1daae2ee05
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-94490/f12ddfcb6a1b523c4523c2fac8ac5e1daae2ee05)
* [Documentation Changes](https://kibana_94490.docs-preview.app.elstc.co/diff)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/35af8a941a68736d82d3c72fe06bb80775a4f828) | [after](https://github.com/elastic/kibana/commit/f12ddfcb6a1b523c4523c2fac8ac5e1daae2ee05) | diff |
| --- | --- | --- | --- |
| `fleet` | 344.7KB | 344.4KB | -324.0B |

### History
* :broken_heart: [Build #114050](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114050/) failed 54ff3ac46f9dc7ee2df690637aa5ed73cbc07bb8
* :broken_heart: [Build #113869](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113869/) failed cc069bffdd5ad935df1eaf68697ddc5d0bec5d54
* :broken_heart: [Build #113863](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113863/) failed e534f709856e97e3d35dec508140f3d2abc6e6fe
* :broken_heart: [Build #113522](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113522/) failed c91092fffdf954b05b7089efe219e9e7de24fbe0
* :broken_heart: [Build #113502](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113502/) failed c2bf522973600a1192a8c485abe68922e5600fe0

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @jen-huang

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113502/","number":"113502","commit":"c2bf522973600a1192a8c485abe68922e5600fe0"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113522/","number":"113522","commit":"c91092fffdf954b05b7089efe219e9e7de24fbe0"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113863/","number":"113863","commit":"e534f709856e97e3d35dec508140f3d2abc6e6fe"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113869/","number":"113869","commit":"cc069bffdd5ad935df1eaf68697ddc5d0bec5d54"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114050/","number":"114050","commit":"54ff3ac46f9dc7ee2df690637aa5ed73cbc07bb8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114322/","number":"114322","commit":"f12ddfcb6a1b523c4523c2fac8ac5e1daae2ee05"}]}
PIPELINE-->

skh reviewed (approved):
I tested this locally and did not manage to break it, so I'd say let's get this in rather sooner than later :rocket: 

I very much like the simplifications in the package installation parts.

jen-huang commented:
Thanks for testing this @skh, appreciate it! I will merge and backport this now.

kibanamachine commented:
## 💚 Backport successful

✅ [7.x](https://github.com/elastic/kibana/pull/95219) / https://github.com/elastic/kibana/pull/95219

This backport PR will be merged automatically after passing CI.
